### PR TITLE
Validate rspectest

### PR DESCRIPTION
**Pull request by katfogy**
**In this milestone,**
- [ ]     I created validations for the models.
- [ ]     I created tests for the models.

### CODE REVIEW

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    end
+  end
+
+  describe 'callbacks' do
+    describe 'after_save' do
+      let(:user) { User.create(name: 'Sigh') }
+      let(:post) { Post.create(title: 'Test Post') }
+    end
+  end
+end
```
Comment (jerryowusu) [inline spec/models/like_spec.rb]: Nice job so far. However, as part of the project requirements, any custom method in the models should have a corresponding unit test. Kindly consider to add a unit test for the `update_like_count` method in the like model here. :pray:

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      expect(association.macro).to eq(:belongs_to)
+      expect(association.options[:class_name]).to eq('User')
+    end
+
+    it 'belongs to a post' do
+      association = described_class.reflect_on_association(:post)
+      expect(association.macro).to eq(:belongs_to)
+    end
+  end
+end
```
Comment (jerryowusu) [inline spec/models/comment_spec.rb]: Nice job so far. However, as part of the project requirements, any custom method in the models should have a corresponding unit test. Kindly consider to add a unit test for the `update_comment_count` method in the comment model here. :pray:

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      post.valid?
+      expect(post.errors[:likes_counter]).to include('is not a number')
+    end
+
+    it 'validates likes_counter is greater than or equal to 0' do
+      post = Post.new(likes_counter: -1)
+      post.valid?
+      expect(post.errors[:likes_counter]).to include('must be greater than or equal to 0')
+    end
+  end
```
Comment (jerryowusu) [inline spec/models/post_spec.rb]: Nice job so far. However, as part of the project requirements, any custom method in the models should have a corresponding unit test. Kindly consider to add a unit test for the `update_posts_count` and `five_most_recent_comments` methods in the post model here. :pray:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @katfogy  

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

noooo


## Highlights
Neat and well organized code :+1: 
Validation tests passing :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @katfogy , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] all test passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nahnahdev** [comment]
## Project Status: APPROVED :green_circle: 
Hi @katfogy , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All project requirements met :heavy_check_mark: 
- Project previously approved :heavy_check_mark: 
- All linter checks passing :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — COMMENTED** [review]
Hi @katfogy , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Added Controller Specs ✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**